presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -281,15 +281,51 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
     @Override
     public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
-        if (isPartitionsSystemTable(tableName)) {
-            return getPartitionsSystemTable(session, tableName);
+        if (partitionsTableHandler.matches(tableName)) {
+            return getPartitionsSystemTable(session, tableName, partitionsTableHandler.getSourceTableName(tableName));
+        }
+        else if (propertiesTableHandler.matches(tableName)) {
+            return getPropertiesSystemTable(session, tableName, propertiesTableHandler.getSourceTableName(tableName));
         }
         return Optional.empty();
     }
 
-    private Optional<SystemTable> getPartitionsSystemTable(ConnectorSession session, SchemaTableName tableName)
+    private Optional<SystemTable> getPropertiesSystemTable(ConnectorSession connectorSession, SchemaTableName tableName, SchemaTableName sourceTableName)
+    {
+        Optional<Table> table = metastore.getTable(sourceTableName.getSchemaName(), sourceTableName.getTableName());
+        if (!table.isPresent() || table.get().getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+            throw new TableNotFoundException(tableName);
+        }
+        Map<String, String> properties = table.get().getParameters();
+        List<ColumnMetadata> columns = properties.keySet().stream().map(key -> new ColumnMetadata(key, VarcharType.VARCHAR)).collect(toImmutableList());
+        ImmutableList<Type> types = columns.stream().map(columnMetadata -> columnMetadata.getType()).collect(toImmutableList());
+        Iterable<List<Object>> propertyValues = singletonList(ImmutableList.copyOf(properties.values()));
```

Comment:
Nit: we normally use `ImmutableList.of()` instead of `singletonList` in the code base. It has the advantage that `ImmutableList.copyOf()` is free. (doesn't matter here, but best to be consistent)

Reply:
done